### DISCUSSION TITLE
Get Product Prices from Prices table based on selected size in a product card using livewire

### DISCUSSION THREAD

**Discussion by rafiqueromeo** [livewire/livewire#6948] [1. Help]
Hi, I am working on an store shop where i have product definition table, sizes table and prices table, 
I have product Card in my client side view where where i have fetch the sizes available for that product in a select element,

1. what i want is to fetch the product prices based on selected size from the prices table , how i will achieve this using 
 livewire version 3?

can somebody help?

**Comment by arthurshlain** [comment]
In your ProductCard Livewire component you can specify `selectedSize` property.

```php
<?php

use Livewire\Component;

class ProductCard extends Component
{
    public mixed $selectedSize;

    public Product $product;

    public function mount(Product $product): void
    {
       $this->product = $product;

       // Set initial size value
       $this->selectedSize = $this->product->getDefaultSize();
    }
   
    ...
}
```

Then bind `selectedSize` with select by using `wire:model` attriubute.

Declare methods `getSizes` and `getPrice` in your Product model to retrieve related data.

```
<div>

    <select wire:model="selectedSize">
        @foreach ($product->getSizes() as $value => $size)
            <option value="{{ $value }}">{{ $size }}</option>
        @endforeach
    </select>

    @if($price = $product->getPrice(['size' => $this->selectedSize ]))
        <p>{{ $price }}</p>
    @else
        <p>Not in stock</p>
    @endif

</div>
```

Call component:
```
<livewire:product-card :product="$product">
```
